### PR TITLE
Redirect non-logged in group + post urls to public post url

### LINKED ISSUE #1503: Fix display of public posts when not logged in and group slug is in the URL
It shows the group landing page, not the post. Works if you take the group slug out of the URL

### PR DESCRIPTION
**Pull request by thomasgwatson**
closes #1503

### CODE REVIEW

File: src/routes/PublicLayoutRouter/PublicLayoutRouter.js [hunk 1]
```diff
@@ -51,6 +51,7 @@ export function PublicGroupDetail (props) {
 
       const result = await dispatch(checkIsPublicGroup(groupSlug))
       const isPublicGroup = result?.payload?.data?.group?.visibility === 2
+      console.log(isPublicGroup, result, 'public group detail')
```
Comment (tibetsprague) [inline src/routes/PublicLayoutRouter/PublicLayoutRouter.js:54]: remove console.log

File: src/routes/PublicLayoutRouter/PublicLayoutRouter.js [hunk 2]
```diff
@@ -64,12 +65,50 @@ export function PublicGroupDetail (props) {
...
 }
 
+export function PublicPostRouteRedirector (props) {
+  const dispatch = useDispatch()
+  const routeParams = useParams()
+  const [loading, setLoading] = useState(true)
+  const history = useHistory()
+  const location = useLocation()
+  const postId = routeParams?.postId
+  console.log(postId, 'ahahhaha')
```
Comment (tibetsprague) [inline src/routes/PublicLayoutRouter/PublicLayoutRouter.js:90]: console.log

File: src/routes/PublicLayoutRouter/PublicLayoutRouter.js [hunk 3]
```diff
@@ -64,12 +65,50 @@ export function PublicGroupDetail (props) {
...
-    <div styleName='center-column non-map-view' id={CENTER_COLUMN_ID}>
-      <GroupDetail {...props} />
-    </div>
+    <>
+      <div styleName='center-column non-map-view' id={CENTER_COLUMN_ID}>
+        <GroupDetail {...props} />
+      </div>
+      <div />
+      <Route
+        path={`(.*)/${POST_DETAIL_MATCH}`}
```
Comment (tibetsprague) [inline src/routes/PublicLayoutRouter/PublicLayoutRouter.js:74]: we might want to match something after the post details too? like if someone copies an edit link?

**Review by tibetsprague — APPROVED** [review]

File: src/routes/PublicLayoutRouter/PublicLayoutRouter.js (hunk 3 above)
Comment (thomasgwatson) [reply src/routes/PublicLayoutRouter/PublicLayoutRouter.js:71]: The path match is not `exact`, so it should catch anything after this match as well

**Pull request merged**